apps/judicial-system/backend/src/app/modules/file/test/limitedAccessFileController/createCaseFileGuards.spec.ts

```diff
@@ -4,32 +4,27 @@ import {
   restrictionCases,
 } from '@island.is/judicial-system/types'
 
+import { verifyGuards } from '../../../../test'
 import { CaseTypeGuard, CaseWriteGuard } from '../../../case'
 import { LimitedAccessWriteCaseFileGuard } from '../../guards/limitedAccessWriteCaseFile.guard'
 import { LimitedAccessFileController } from '../../limitedAccessFile.controller'
 
 describe('LimitedAccessFileController - Create case file guards', () => {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  let guards: any[]
-
-  beforeEach(() => {
-    guards = Reflect.getMetadata(
-      '__guards__',
-      LimitedAccessFileController.prototype.createCaseFile,
-    )
-  })
-
-  it('should have the right guard configuration', () => {
-    expect(guards).toHaveLength(3)
-    expect(guards[0]).toBeInstanceOf(CaseTypeGuard)
-    expect(guards[0]).toEqual({
-      allowedCaseTypes: [
-        ...restrictionCases,
-        ...investigationCases,
-        ...indictmentCases,
-      ],
-    })
-    expect(new guards[1]()).toBeInstanceOf(CaseWriteGuard)
-    expect(new guards[2]()).toBeInstanceOf(LimitedAccessWriteCaseFileGuard)
-  })
+  verifyGuards(
+    LimitedAccessFileController,
+    'createCaseFile',
+    [CaseTypeGuard, CaseWriteGuard, LimitedAccessWriteCaseFileGuard],
+    [
+      {
+        guard: CaseTypeGuard,
+        prop: {
+          allowedCaseTypes: [
```

Comment:
Do we need this allowCaseTypes? Doesn't this cover all case types?

Reply:
Strictly speaking we don't. However, I was not addressing that in this PR. Also, I would argue we should have this guard everywhere so that when we add new case type we do not automatically grant everyone access to endpoints that perhaps should be restricted. In this case I would err on the side of overly cautious.